src/fwup_apply.c

```diff
@@ -416,7 +416,7 @@ int fwup_apply(const char *fw_filename, const char *task_prefix, int output_fd,
     OK_OR_CLEANUP(fatfs_ptr_callback(&fctx, -1, NULL));
 
     // Flush a subarchive that's being built.
-    OK_OR_CLEANUP(subarchive_ptr_callback(&fctx, "", NULL, NULL));
+    OK_OR_CLEANUP(subarchive_ptr_callback(&fctx, NULL, NULL, NULL));
```

Comment:
I didn't totally understand what this was trying to do or why it worked correctly on Linux and not Windows, but what I noticed was that inside the function, it was checked whether the second parameter was 'truthy' in an `if` statement, so maybe the mingw compiler thinks an empty string counts as true where Linux GCC doesn't? I figured NULL would definitely be 'falsey', so I switched it to that since it seemed like that was the intent. It might be causing one or more tests to fail. I'm not sure exactly why the remaining ones fail yet.